### PR TITLE
Modus NavBar - Include Add icon

### LINKED ISSUE #1247: Modus NavBar - Include Add icon
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Proposal

Include the Add icon in the NavBar component.

image

It is e-Builder's intent to work on this request.

### Motivation and context

This would be leveraged to provide quick add functionality.  This would enable our clients to always have access to our quick add feature in a convenient location.

### Which npm package?

@trimble-oss/modus-web-components

### Priority

Medium

### What is the product name the feature is requested for?

e-Builder

### What is the team name the product belongs to?

e-Builder

### What is the division name the team belongs to?

e-Builder

### PR DESCRIPTION
**Pull request by cjwinsor**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1247

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

We tested both using the index.html page and running storybook to verify. Review was done by different developer than the one doing the work.  

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (including CHANGELOG updates)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-navbar/modus-navbar.e2e.ts [hunk 1]
```diff
@@ -114,4 +114,31 @@ describe('modus-navbar', () => {
...
+    const page = await newE2EPage();
+    await page.setContent('<modus-navbar show-add></modus-navbar>');
+
+    const addMenuOpen = await page.spyOnEvent('addMenuOpen');
+    await page.waitForChanges();
+    const addMenuButton = await page.find('modus-navbar >>> [data-test-id="add-menu"]');
+    await addMenuButton.click({ clickCount: 2 });
+    await page.waitForChanges();
+    expect(addMenuOpen).toHaveReceivedEventTimes(1);
+  });
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/modus-navbar.e2e.ts:128]: Similar to other tests, checks for proper emitting of event

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx [hunk 2]
```diff
@@ -270,137 +302,154 @@ export class ModusNavbar {
...
+                </span>
+                {this.notificationsMenuVisible && (
+                  <modus-navbar-notifications-menu reverse={this.reverse}>
+                    <slot name={this.SLOT_NOTIFICATIONS}></slot>
+                  </modus-navbar-notifications-menu>
+                )}
+              </div>
+            )}
+            {this.showPendoPlaceholder && <div class={'pendo-placeholder'} />}
+            {this.showAdd && (
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/modus-navbar.tsx:367]: This is where we handled `showAdd`

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx [hunk 3]
```diff
@@ -4,6 +4,16 @@ import { html } from 'lit-html';
...
+  argTypes: {
+    showAdd: {
+      name: 'show-add',
+      description: 'Toogle the plus button',
+      table: {
+        defaultValue: { summary: false },
+        type: { summary: 'boolean' },
+      },
+    },
+  },
```
Comment (cjwinsor) [inline stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx:16]: Allowing canvas to toggle this on/off.  Will most likely tackle the rest of the properties in future PRs

**Comment by cjwinsor** [comment]
We can wait on this being included in a future release at this time.

File: stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx [hunk 4]
```diff
@@ -4,6 +4,16 @@ import { html } from 'lit-html';
 
 export default {
   title: 'Components/Navbar',
+  argTypes: {
+    showAdd: {
+      name: 'show-add',
+      description: 'Toogle the plus button',
```
Comment (coliff) [inline stencil-workspace/storybook/stories/components/modus-navbar/modus-navbar.stories.tsx:10]: Can you pls fix this typo on 'toggle'

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
As discussed we needed an array input for icons.

**Comment by cjwinsor** [comment]
Closing, will create new PR when work is completed on changes.

**Pull request closed without merging**